src/waveform/renderers/glwaveformrendererrgb.cpp

```diff
@@ -165,16 +189,30 @@ void GLWaveformRendererRGB::draw(QPainter* painter, QPaintEvent* /*event*/) {
             for (int visualIndex = firstIndex;
                     visualIndex < lastIndex;
                     visualIndex += 2) {
-
-                float low  = lowGain  * (float) math_max(data[visualIndex].filtered.low,  data[visualIndex+1].filtered.low);
-                float mid  = midGain  * (float) math_max(data[visualIndex].filtered.mid,  data[visualIndex+1].filtered.mid);
-                float high = highGain * (float) math_max(data[visualIndex].filtered.high, data[visualIndex+1].filtered.high);
+                float low = lowGain *
+                        static_cast<float>(
+                                math_max(data[visualIndex].filtered.low,
+                                        data[visualIndex + 1].filtered.low));
+                float mid = midGain *
+                        static_cast<float>(
+                                math_max(data[visualIndex].filtered.mid,
+                                        data[visualIndex + 1].filtered.mid));
+                float high = highGain *
+                        static_cast<float>(
+                                math_max(data[visualIndex].filtered.high,
+                                        data[visualIndex + 1].filtered.high));
 
                 float all = sqrtf(low * low + mid * mid + high * high) * kHeightScaleFactor;
 
-                float red   = low * m_rgbLowColor_r + mid * m_rgbMidColor_r + high * m_rgbHighColor_r;
-                float green = low * m_rgbLowColor_g + mid * m_rgbMidColor_g + high * m_rgbHighColor_g;
-                float blue  = low * m_rgbLowColor_b + mid * m_rgbMidColor_b + high * m_rgbHighColor_b;
+                float red = low * static_cast<float>(m_rgbLowColor_r) +
+                        mid * static_cast<float>(m_rgbMidColor_r) +
+                        high * static_cast<float>(m_rgbHighColor_r);
+                float green = low * static_cast<float>(m_rgbLowColor_g) +
```

Comment:
It might be a good idea to break out the GL related changes into yet one more separate PR.  It seems we can use some unifying of styles and types in our various renderers

Reply:
There are still more issues I want to fix in all of our codebase, so I'd rather merge it as is for now. Feel free to open a follow-up PR that unifies the GL stuff.